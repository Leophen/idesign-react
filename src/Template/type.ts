export interface TemplateProps {
  /**
   * 类名
   */
  className?: string;
  /**
   * 内容
   */
  children?: React.ReactNode;
  /**
   * 自定义样式
   */
  style?: React.CSSProperties;
}
