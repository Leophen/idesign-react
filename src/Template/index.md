---
nav:
  title: 组件
  path: /components
group:
  title: XX组件
  order: 6
---

# Template XX

## Demo

```tsx
import React from 'react';
import { Template } from 'idesign-react';

const App = () => {
  return (
    <div className="idesign-demo-block-column">
      <div className="idesign-demo-block-row">
        <Template>填充按钮</Template>
      </div>
    </div>
  );
};

export default App;
```
