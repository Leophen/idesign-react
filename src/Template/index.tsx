import React from 'react';
import classNames from 'classnames';
import './index.scss';
import { TemplateProps } from './type';

const Template: React.FC<TemplateProps> = (props) => {
  const {
    children,
    className,
    style,
    ...restProps
  } = props;

  return (
    <div
      className={classNames(
        'i-template',
        className
      )}
      style={{ ...style }}
      {...restProps}
    >
      {children}
    </div>
  );
};

Template.displayName = 'Template';

export default Template;
