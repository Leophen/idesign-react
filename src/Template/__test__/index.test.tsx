import { render } from '@testing-library/react';
import Template from '../index';

describe('Template 组件测试', () => {
  it('create', () => {
    const { asFragment } = render(<Template />);
    expect(asFragment()).toMatchSnapshot();
  });
});

