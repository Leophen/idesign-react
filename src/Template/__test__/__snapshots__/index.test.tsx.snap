// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template 组件测试 create 1`] = `
<DocumentFragment>
  <div
    class="i-template"
  />
</DocumentFragment>
`;
